filebeat/fileset/fileset.go

```diff
@@ -0,0 +1,255 @@
+		}
+	}
+
+	// overrides from the config
+	for name, val := range fs.fcfg.Var {
+		vars[name] = val
+	}
+
+	return vars, nil
+}
+
+// resolveVariable considers the value as a template so it can refer to built-in variables
+// as well as other variables defined before them.
+func resolveVariable(vars map[string]interface{}, value interface{}) (interface{}, error) {
+	switch v := value.(type) {
+	case string:
+		return applyTemplate(vars, v)
+	case []interface{}:
+		transformed := []interface{}{}
+		for _, val := range v {
+			s, ok := val.(string)
+			if ok {
+				transf, err := applyTemplate(vars, s)
+				if err != nil {
+					return nil, fmt.Errorf("array: %v", err)
+				}
+				transformed = append(transformed, transf)
+			} else {
+				transformed = append(transformed, val)
+			}
+		}
+		return transformed, nil
+	}
+	return value, nil
+}
+
+// applyTemplate applies a Golang text/template
+func applyTemplate(vars map[string]interface{}, templateString string) (string, error) {
+	tpl, err := template.New("test").Parse(templateString)
+	if err != nil {
+		return "", fmt.Errorf("Error parsing template %s: %v", templateString, err)
+	}
+	buf := bytes.NewBufferString("")
+	err = tpl.Execute(buf, vars)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+// getBuiltinVars computes the supported built in variables and groups them
+// in a dictionary
+func (fs *Fileset) getBuiltinVars() (map[string]interface{}, error) {
+	host, err := os.Hostname()
+	if err != nil || len(host) == 0 {
+		return nil, fmt.Errorf("Error getting the hostname: %v", err)
+	}
+	split := strings.SplitN(host, ".", 2)
+	hostname := split[0]
+	domain := ""
+	if len(split) > 1 {
+		domain = split[1]
+	}
+
+	return map[string]interface{}{
+		"hostname": hostname,
+		"domain":   domain,
+	}, nil
+}
+
+func (fs *Fileset) getProspectorConfig() (*common.Config, error) {
+	path, err := applyTemplate(fs.vars, fs.manifest.Prospector)
+	if err != nil {
+		return nil, fmt.Errorf("Error expanding vars on the prospector path: %v", err)
+	}
+	contents, err := ioutil.ReadFile(filepath.Join(fs.modulePath, fs.name, path))
+	if err != nil {
+		return nil, fmt.Errorf("Error reading prospector file %s: %v", path, err)
+	}
+
+	yaml, err := applyTemplate(fs.vars, string(contents))
+	if err != nil {
+		return nil, fmt.Errorf("Error interpreting the template of the prospector: %v", err)
+	}
+
+	cfg, err := common.NewConfigWithYAML([]byte(yaml), "")
+	if err != nil {
+		return nil, fmt.Errorf("Error reading prospector config: %v", err)
+	}
+
+	// overrides
+	if len(fs.fcfg.Prospector) > 0 {
+		overrides, err := common.NewConfigFrom(fs.fcfg.Prospector)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating config from prospector overrides: %v", err)
+		}
+		cfg, err = common.MergeConfigs(cfg, overrides)
+		if err != nil {
+			return nil, fmt.Errorf("Error applying config overrides: %v", err)
+		}
+	}
+
+	return cfg, nil
```

Comment:
a debug log entry here outputting the final merged config would be nice to have.